web/js/blocks/cw-lists.js

```diff
@@ -14,835 +14,846 @@
-                'blocklyTextEmph'));
-        this.appendValueInput('VAR0')
-            .setAlign(Blockly.ALIGN_RIGHT)
-            .appendField(new Blockly.FieldLocalVar(this.vars_[0], this.getArgType(
-                0)))
-            .appendField('\u2190');
-        this.appendValueInput('VAR1')
-            .setAlign(Blockly.ALIGN_RIGHT)
-            .appendField(new Blockly.FieldLocalVar(this.vars_[1], this.getArgType(
-                1)))
-            .appendField('\u2190');
-        this.setOutput(true);
-        this.setMutator(new Blockly.Mutator(['lists_comp_var',
-            'lists_comp_guard'
-        ]));
-        this.setTooltip(Blockly.Msg.LISTS_CREATE_WITH_TOOLTIP);
-        this.varCount_ = 2;
-        this.guardCount_ = 0;
-        this.resetArrows();
-        Blockly.TypeInf.defineFunction('&&&', Type.fromList([Type.Lit(
-            'Truth'), Type.Lit('Truth'), Type.Lit('Truth')]));
-        Blockly.TypeInf.defineFunction('filtB', Type.fromList([Type.Var('a'),
-            Type.Lit('Truth'), Type.Var('a')
-        ]));
-        Blockly.TypeInf.defineFunction('returnl', Type.fromList([Type.Var(
-            'a'), Type.Lit('list', [Type.Var('a')])]));
-        Blockly.TypeInf.defineFunction('froml', Type.fromList([Type.Lit(
-            'list', [Type.Var('a')]), Type.Var('a')]));
-        Blockly.TypeInf.defineFunction('bindl', Type.fromList(
-            [Type.Lit('list', [Type.Var('a')]), Type.fromList([Type
-                .Var('a'), Type.Lit('list', [Type.Var('b')])
-            ]), Type.Lit('list', [Type.Var('b')])]));
-
-    },
-
-    foldr1(fn, xs) {
-        let result = xs[xs.length - 1];
-        for (let i = xs.length - 2; i > -1; i--) {
-            result = fn(xs[i], result);
-        }
-        return result;
-    },
-
-    getExpr() {
-
-        // Do main exp
-        let mainExp = Exp.Var('undef');
-        if (this.getInput('DO').connection.isConnected()) {
-            mainExp = this.getInput('DO').connection.targetBlock().getExpr();
-        }
-
-        mainExp.tag = this.getInput('DO').connection;
-        mainExp = Exp.App(Exp.Var('returnl'), mainExp);
-
-        // Do Guards
-        const guardExps = [];
-        for (let i = 0; i < this.guardCount_; i++) {
-            const inp = this.getInput(`GUARD${i}`);
-            if (inp.connection && inp.connection.isConnected()) {
-                const exp = inp.connection.targetBlock().getExpr();
-                exp.tag = inp.connection;
-                guardExps.push(exp);
-            } else {
-                const exp = Exp.Var('undef');
-                exp.tag = inp.connection;
-                guardExps.push(exp);
-            }
-        }
+'use strict'
+
+goog.provide('Blockly.Blocks.cwLists')
+
+goog.require('Blockly.Blocks')
+
+const listsHUE = 260
+
+Blockly.Blocks.lists_comprehension = {
+  /**
+   * Block for creating a list with any number of elements of any type.
+   * @this Blockly.Block
+   */
+  init () {
```

Comment:
What's with these spaces?  It's not even consistent; it's adding a space at definition, but not at function application.  Would definitely prefer to get rid of these.

Reply:
I don't think I understand the reasoning behind this rule either. Good news is that it could be turned off.